### ISSUE TITLE
Scoping issue with const/let and switch statement

### ISSUE THREAD

**Issue by assaf** [6to5/6to5#564]
Minimal test case:

```js
function fail() {
  if (true) {
    const y = 1;
    switch (1) {
      case 1: {
        function x() { return y };
        break;
      }
    }
  }
}
```

The error I get is:

```
        if (_ret === "break") break _loop;
                                    ^^^^^
SyntaxError: Undefined label '_loop'
```

This happens when:
- Inside a `case` statement
- That ends with a `break`
- And contains a function that accesses a variable in the enclosing scope
- Which is not the same scope as the function (i.e. scoped to condition, for loop, try/catch)
- And is defined with local scope (`const` or `let`)

**Comment by sebmck** [comment]
Looks like the block scoping transformer assumes everything containing a `break` statement is a loop. A test case that I can add to the suite would be great, basically something like [this](https://github.com/6to5/6to5/blob/master/test/fixtures/transformation/es6-let-scoping/exec-duplicate-in-upper-scope/exec.js) or [this](https://github.com/6to5/6to5/blob/master/test/fixtures/transformation/es6-let-scoping/exec-for-loop-head/exec.js).

**Comment by sebmck** [comment]
Fixed in the 3.0.0 branch

**Comment by sebmck** [comment]
Fixed as of 3.0.0
